src/Ssh.php

```diff
@@ -111,7 +111,7 @@ protected function getExtraOptions(): string
             $extraOptions[] = "-i {$this->pathToPrivateKey}";
         }
 
-        if ($this->port) {
+        if (null !== $this->port) {
```

Comment:
A port can not be a negative integer ... the `usePort`  method should also be changed to throw exception for negative integer